modules/xfem/src/userobjects/InterfaceMeshCut2DUserObject.C

```diff
@@ -0,0 +1,267 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#include "InterfaceMeshCut2DUserObject.h"
+#include "XFEMMovingInterfaceVelocityBase.h"
+
+registerMooseObject("XFEMApp", InterfaceMeshCut2DUserObject);
+
+InputParameters
+InterfaceMeshCut2DUserObject::validParams()
+{
+  InputParameters params = InterfaceMeshCutUserObjectBase::validParams();
+  params.addClassDescription(
+      "Creates a UserObject for a mesh cutter in 2D material interface problems");
+  return params;
+}
+
+InterfaceMeshCut2DUserObject::InterfaceMeshCut2DUserObject(const InputParameters & parameters)
+  : InterfaceMeshCutUserObjectBase(parameters)
+{
+  for (const auto & elem : _cutter_mesh->element_ptr_range())
```

Comment:
Is there a way to tell element types without looping through all elements? I would guess there is. But this is not a big deal.

Reply:
I could not find it. :-(